compiler/options.nim

```diff
@@ -231,6 +231,8 @@ type
     foreignPackageNotes*: TNoteKinds
     notes*: TNoteKinds
     mainPackageNotes*: TNoteKinds
+    cmdLineNotes*: TNoteKinds
```

Comment:
Why do we need these?

Reply:
I found a cleaner way, and also fixed a number of issues, will explain below in comments